### PR TITLE
feat: add possibility to select custom Ollama models

### PR DESCRIPTION
**Pull request by Shanorino**

### CODE REVIEW

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx [hunk 1]
```diff
@@ -109,40 +113,24 @@ export const OllamaForm = ({
...
-      ) : getLLMModels(models ?? []).length === 0 ? (
-        <FormMessage>
-          There is no LLM model available using Ollama. <br />
-          Please pull a Ollama LLM model from &nbsp;
-          <a href="https://ollama.com/library" target="_blank" rel="noreferrer">
-            https://ollama.com/library
-          </a>
-        </FormMessage>
-      ) : getEmbeddingModels(models ?? []).length == 0 ? (
+      ) : (
```
Comment (marcusschiesser) [inline src/ragapp/admin-ui/sections/config/providers/ollama.tsx:116]: don't remove the LLM model check

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx [hunk 2]
```diff
@@ -109,40 +113,24 @@ export const OllamaForm = ({
...
         <>
           <ModelForm
             form={form}
+            title="LLM Model"
+            description="Select an LLM model to use for language generation."
             defaultValue={defaultValues.model}
             supportedModels={getLLMModels(models ?? [])}
           />
-          <FormMessage>
```
Comment (marcusschiesser) [inline src/ragapp/admin-ui/sections/config/providers/ollama.tsx:127]: re-add the embedding model check, but change text to:
```
One of the embedding models <i>${embeddingModels.join(",")}</i> is required. Please pull an Ollama embedding model (e.g. 'nomic-embed-text') from &nbsp;
          <a href="https://ollama.com/library" target="_blank" rel="noreferrer">
            https://ollama.com/library
          </a>
```
```

**Review by marcusschiesser — COMMENTED** [review]
Thanks @Shanorino - Looking good, please re add the model checks

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx (hunk 2 above)
Comment (Shanorino) [reply src/ragapp/admin-ui/sections/config/providers/ollama.tsx:127]: I've added the check for embedding, meanwhile I kept the LLM form if embedding model is missing. It seems to be working as expected.

image

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx [hunk 3]
```diff
@@ -109,40 +117,53 @@ export const OllamaForm = ({
...
             https://ollama.com/library
           </a>
         </FormMessage>
-      ) : getEmbeddingModels(models ?? []).length == 0 ? (
+      ) : embeddingModelsAvailable.length === 0 ? (
         <>
           <ModelForm
             form={form}
+            title="LLM Model"
+            description="Select an LLM model to use for language generation."
```
Comment (marcusschiesser) [inline src/ragapp/admin-ui/sections/config/providers/ollama.tsx:133]: why did you remove these lines?

Comment (Shanorino) [reply src/ragapp/admin-ui/sections/config/providers/ollama.tsx:133]: I think they are added for better description, not removed

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx [hunk 4]
```diff
@@ -109,40 +117,53 @@ export const OllamaForm = ({
...
-        <ModelForm
-          form={form}
-          defaultValue={defaultValues.model}
-          supportedModels={getLLMModels(models ?? [])}
-        />
+        <>
+          <ModelForm
+            form={form}
+            title="LLM Model"
+            description="Select an LLM model to use for language generation."
```
Comment (marcusschiesser) [inline src/ragapp/admin-ui/sections/config/providers/ollama.tsx:154]: ah i see, there are defaults, i think then better remove it here:
```suggestion
```

**Review by marcusschiesser — APPROVED** [review]

**Review by marcusschiesser — APPROVED** [review]
Thanks @Shanorino

**Comment by marcusschiesser** [comment]
@Shanorino sorry, can you check the linter please - once the checks pass we can merge this PR

**Comment by Shanorino** [comment]
> @Shanorino sorry, can you check the linter please - once the checks pass we can merge this PR

Sure, I can take a look later and will let you know

**Comment by Shanorino** [comment]
Hi @marcusschiesser , I’m not sure if you’ve seen the latest commit. I ran the linting command with “—write” option as the GitHub action suggested. Can you try again and see if the pipeline runs through now?

File: src/ragapp/admin-ui/sections/config/providers/ollama.tsx [hunk 5]
```diff
@@ -109,40 +114,52 @@ export const OllamaForm = ({
...
             https://ollama.com/library
           </a>
         </FormMessage>
-      ) : getEmbeddingModels(models ?? []).length == 0 ? (
+      ) : embeddingModelsAvailable.length === 0 ? (
         <>
           <ModelForm
             form={form}
+            title="LLM Model"
+            description="Select an LLM model to use for language generation."
```
Comment (marcusschiesser) [inline src/ragapp/admin-ui/sections/config/providers/ollama.tsx:130]:
```suggestion
```
it makes sense removing it
